 design/detailed-diagrams.md

```diff
@@ -0,0 +1,686 @@
+end
+
+subgraph Java
+  JavaESDK["ESDK<sub>java</sub>"]
+  JavaCrypto["Crypto<sub>java</sub>"]
+  JavaMaterialProviders["JavaMaterialProviders<sub>java</sub>"]
+
+  Application -->|Unverified| JavaESDK
+  Application -->|Unverified| JavaMaterialProviders
+
+  JavaESDK -->|Verified| JavaCrypto
+  JavaESDK -->|Verified| JavaMaterialProviders
+  JavaMaterialProviders -->|Verified| JavaCrypto
+end
+
+Dafny --> Java
+
+```
+
+Now let us return to our first example
+and model the dependencies outside the translated program.
+In this way when the Dafny is translated
+the dependency information is not lost.
+This will require coordinating dependencies
+between the Dafny and the Native runtime.
+
+```mermaid
+---
+title: Dafny with dependencies expressed in native
+---
+graph LR
+
+subgraph Dafny
+
+  A -->|Dependency| B
+  C -->|Dependency| B
+end
+
+subgraph Java
+  JavaA["A<sub>java</sub>"]
+  JavaC["C<sub>java</sub>"]
+  JavaB["B<sub>java</sub>"]
+
+  JavaA -->|Dependency| JavaB
+  JavaC -->|Dependency| JavaB
+end
+
+subgraph Net[".Net"]
+  NetA["A<sub>.net</sub>"]
+  NetB["B<sub>.net</sub>"]
+  NetC["C<sub>.net</sub>"]
+
+  NetA -->|Dependency| NetB
+  NetC -->|Dependency| NetB
+end
+
+subgraph Js["Javascript"]
+  JsA["A<sub>js</sub>"]
+  JsB["B<sub>js</sub>"]
+  JsC["C<sub>js</sub>"]
+
+  JsA -->|Dependency| JsB
+  JsC -->|Dependency| JsB
+end
+
+Dafny ---> Java
+Dafny ---> Net
+Dafny ---> Js
+```
+
+
+## Smithy Dafny
+
+### Library Development
+
+Here is a example of the various components
+of smithy dafny fit together to translate a library.
+
+
+```mermaid
+---
+title: How smithy-dafny packages a Dafny Implementation
+---
+graph LR
+
+subgraph Smithy
+  AModel["Model for A"]
+end
+
+subgraph Dafny
+  subgraph A["Package for A"]
+    ATypes["Dafny Types for A"]
+    AInterface["Dafny Interface for A"]
+    AImplementation["Dafny implementation of A"]
+
+    AImplementation --> AInterface
+    ATypes --> AInterface
+    ATypes --> AImplementation
+
+    AModel -.->|CodeGen| ATypes
+    AModel -.->|CodeGen| AInterface
+  end
+
+  subgraph ATests["Tests for A"]
+    AUnitTests["Unit Tests for A"]
+    AIntegrationTests["Integration Tests for A"]
+
+    AUnitTests --> |Testing| AImplementation
+    AIntegrationTests -->|Testing| AInterface
+  end
+end
+
+subgraph Net[".Net"]
+  subgraph NetA["A<sub>.net</sub>"]
+    NetAFromDafny
+    NetATypes
+    NetAInterface
+    NetATypeConversion["To/From Dafny"]
+
+    NetATypeConversion --> NetAInterface
+    NetAFromDafny --> NetAInterface
+    NetATypes --> NetAInterface
+    NetATypes --> NetATypeConversion
+    AModel -.->|CodeGen| NetATypes
+    AModel -.->|CodeGen| NetAInterface
+    AModel -.->|CodeGen| NetATypeConversion
+    A ==>|Dafny Compilation| NetAFromDafny
+  end
+
+  NetATests
+
+  ATests ==>|Dafny Compilation| NetATests
+  NetATests -->|Testing| NetAFromDafny
+end
+```
+
+### Dependencies
+
+This is a description of how Smithy Dafny
+handles dependencies.
```

Comment:
It would help to explain exactly what we are seeing here. It looks like we have two packages, A and B, both are smithy-dafny projects, and B depends on A.

Reply:
done